tests/unit-tests/enrolment/test-class-sensei-enrolment-provider-journal-store.php

```diff
@@ -0,0 +1,274 @@
+		$user   = $this->factory->user->create();
+		$this->enableJournal();
+
+		$provider_results = [
+			'manual' => true,
+			'simple' => false,
+		];
+
+		Sensei_Enrolment_Provider_Journal_Store::register_possible_enrolment_change( $provider_results, $user, $course );
+		$after_first_change = microtime( true );
+		usleep( 10000 );
+
+		$provider_results = [
+			'memberships' => false,
+		];
+
+		Sensei_Enrolment_Provider_Journal_Store::register_possible_enrolment_change( $provider_results, $user, $course );
+		$after_second_change = microtime( true );
+		usleep( 10000 );
+
+		$provider_results = [
+			'manual'      => false,
+			'memberships' => true,
+		];
+
+		Sensei_Enrolment_Provider_Journal_Store::register_possible_enrolment_change( $provider_results, $user, $course );
+
+		$first_change_snapshot = Sensei_Enrolment_Provider_Journal_Store::get_enrolment_snanpshot( $user, $course, $after_first_change );
+		$this->assertCount( 2, $first_change_snapshot, 'There should be 2 providers in the snapshot.' );
+		$this->assertTrue( $first_change_snapshot['manual'], 'The manual provider status should be true.' );
+		$this->assertFalse( $first_change_snapshot['simple'], 'The simple provider status should be false.' );
+
+		$second_change_snapshot = Sensei_Enrolment_Provider_Journal_Store::get_enrolment_snanpshot( $user, $course, $after_second_change );
+		$this->assertCount( 1, $second_change_snapshot, 'There should be 1 provider in the snapshot.' );
+		$this->assertFalse( $second_change_snapshot['memberships'], 'The memberships provider status should be false.' );
+
+		$current_snapshot = Sensei_Enrolment_Provider_Journal_Store::get_enrolment_snanpshot( $user, $course );
+		$this->assertCount( 2, $current_snapshot, 'There should be 2 providers in the snapshot.' );
+		$this->assertTrue( $current_snapshot['memberships'], 'The memberships provider status should be true.' );
+		$this->assertFalse( $current_snapshot['manual'], 'The manual provider status should be false.' );
+	}
+
+	/**
+	 * Tests that Sensei_Enrolment_Provider_Journal_Store::get_enrolment_snanpshot returns correct snapshots after the
+	 * timestamps are rounded.
+	 */
+	public function testHistorySnapshotsRoundings() {
+		$course = $this->factory->course->create();
+		$user   = $this->factory->user->create();
+		$this->enableJournal();
+
+		$method = new ReflectionMethod( Sensei_Enrolment_Provider_Journal_Store::class, 'get' );
+		$method->setAccessible( true );
+		$journal_store = $method->invoke( null, $user, $course );
+
+		$journal_json = '
+			{
+				"manual": {
+				"h": [
+						{
+							"t": 1586530073.434888,
+							"s": false
+						},
+						{
+							"t": 1586530073.432448,
+							"s": true
+						}
+					],
+					"l": []
+				},
+				"memberships": {
+				"h": [
+						{
+							"t": 1586530073.434444,
+							"s": null
+						},
+						{
+							"t": 1586530073.431507,
+							"s": false
+						}
+					],
+					"l": []
+				}
+			}';
+
+
```

Comment:
This double space is the PHPCS issue failing builds.

Reply:
Fixed in 8d2f10a0076effec90625a344c3468b8ab352eef